sdk/nodejs/helm/v3/helm.ts

```diff
@@ -146,137 +141,86 @@ export class Chart extends yaml.CollectionComponentResource {
-                        nodepath.join(chartDir.name, fetchedChartName, "values.yaml")
-                    );
-                } else {
-                    chart = cfg.path;
-                    defaultValues = path.quotePath(nodepath.join(chart, "values.yaml"));
-                }
-
-                // Write overrides file.
-                const data = JSON.stringify(cfg.values || {}, undefined, "  ");
-                fs.writeFileSync(overrides.name, data);
-
-                // Does not require Tiller. From the `helm template` documentation:
-                //
-                // >  Render chart templates locally and display the output.
-                // >
-                // > This does not require Tiller. However, any values that would normally be
-                // > looked up or retrieved in-cluster will be faked locally. Additionally, none
-                // > of the server-side testing of chart validity (e.g. whether an API is supported)
-                // > is done.
-                const release = shell.quote([releaseName]);
-                const values = path.quotePath(overrides.name);
-                const apiVersionsArgs = cfg.apiVersions
-                    ? cfg.apiVersions.length > 1
-                        ? `--api-versions={${cfg.apiVersions
-                            .map(apiVersion => shell.quote([apiVersion]))
-                            .join(',')}}`
-                        : `--api-versions=${shell.quote(cfg.apiVersions)}`
-                    : '';
-                const namespaceArg = cfg.namespace
-                    ? `--namespace ${shell.quote([cfg.namespace])}`
-                    : "";
-
-                // Check the helm version - v2 or v3
-                let helmVerCmd = `helm version --short || true`;
-                var helmVer;
-                try {
-                    helmVer = execSync(
-                        helmVerCmd,
-                        {
-                            stdio: ['pipe', 'pipe', 'ignore'], // Suppress tiller version error
-                        },
-                    ).toString();
-                } catch (e) {
-                    helmVer = e.stdout.toString();
-                }
+            return this.parseChart(cfg, releaseName, opts)
+        });
+    }
 
-                cmd = `helm template ${chart} --name-template ${release} --values ${defaultValues} --values ${values} ${apiVersionsArgs} ${namespaceArg}`;
-                // Helm v2 returns version like this:
-                // Client: v2.16.7+g5f2584f
-                // Helm v3 returns a version like this:
-                // v3.1.2+gd878d4d
-                // --include-crds is available in helm v3.1+ so check for a regex matching that version
-                if (RegExp('^v3\.[1-9]').test(helmVer)) {
-                    cmd += ` --include-crds`
+    parseChart(config: ChartOpts | LocalChartOpts, releaseName: string, opts?: pulumi.ComponentResourceOptions) {
```

Comment:
What's the thought behind repeating `parseChart` here and in the template, but not the rest of the code in the template?

Reply:
I'm not 100% sure I understand the question, but most of this code is no longer needed since this logic happens in the provider invoke now.